src/hooks/useErc20.ts

```diff
@@ -0,0 +1,65 @@
+import { useEffect, useState } from 'react'
+
+import { TokenErc20 } from '@gnosis.pm/dex-js'
+
+import { Network } from 'types'
+
+import { buildErc20Key, saveSingleErc20 } from 'reducers-actions/erc20'
+
+import useGlobalState from 'hooks/useGlobalState'
+
+import { getErc20Info } from 'services/helpers'
+
+import { web3, erc20Api } from 'apps/explorer/api'
+
+import { ExplorerAppState } from 'apps/explorer/state'
+
+type Params = {
+  address?: string
+  networkId?: Network
+}
+
+type Return = { isLoading: boolean; error?: string; value?: TokenErc20 }
+
+export function useErc20(params: Params): Return {
+  const { address, networkId } = params
+
+  const [isLoading, setIsLoading] = useState(false)
+  const [error, setError] = useState('')
+
+  const [{ erc20s }, dispatch] = useGlobalState<ExplorerAppState>()
+
+  // Initial state from global state
+  const [erc20, setErc20] = useState<TokenErc20 | undefined>(
+    address && networkId ? erc20s.get(buildErc20Key(networkId, address)) : undefined,
+  )
+
+  useEffect(() => {
+    // Only try to fetch it if not on global state
+    if (!erc20 && address && networkId) {
```

Comment:
I would try to reduce the complexity of this functions

Can we just, for example, instead of wrapping all in a if, do a `if(the opposite) return undefined`
Then u reduce all the cyclomatic complexity by 1

Not sure if it's worth it, but we can use async/await syntax. For example:

```ts
useEffect(() => {
    async function getErc20AndPersist(): Promise<void> {
      // Only try to fetch it if not on global state
      if (erc20 || !address || !networkId) {
        return undefined
      }

      try {
        const fetchedErc20 = await getErc20Info({ tokenAddress: address, networkId, web3, erc20Api })

        // When not found, it returns null
        if (fetchedErc20) {
          dispatch(saveSingleErc20(fetchedErc20, networkId))
          setErc20(fetchedErc20)
        }
      } catch (error) {
        // Set error only when the call fails
        // Not finding the token is not an error
        const msg = `Failed to fetch erc20 details for ${address} on network ${networkId}`
        console.error(msg, error)
        setError(msg)
      } finally {
        setIsLoading(false)
      }
    }

    getErc20AndPersist()
  }, [address, dispatch, erc20, networkId])
```

Reply:
https://github.com/gnosis/gp-ui/pull/132

Keep in mind this PR depends on https://github.com/gnosis/gp-ui/pull/130